### PR TITLE
Add pvc to prune whitelist

### PR DESCRIPTION
**Pull request by jessie-JNing**
**What are you trying to accomplish with this PR?**
Work on: https://github.com/Shopify/kubernetes-deploy/issues/519
Add PersistentVolumeClaims to Krane's prune whitelist.

If Krane detects PVCs who have `kubectl.kubernetes.io/last-applied-configuration` annotation but no template in repo, Krane will prune it. 
PVCs who deployed with `--no-prune` option can be exempted from pruning.

**How is this accomplished?**
- Add `core/v1/PersistentVolumeClaim` to prune whitelist
- Fix some tests

**What could go wrong?**
if there's missing PVCs that our [safety check](https://github.com/Shopify/cloudplatform/issues/1534) wasn't able to cover, may cause data lost.

### CODE REVIEW

**Comment by jessie-JNing** [comment]
I won't merge it until the safety check https://github.com/Shopify/cloudplatform/issues/1534 is passed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 - Make sure that we only declare a Service of type LoadBalancer as deployed after its IP address is published. [#547](https://github.com/Shopify/kubernetes-deploy/pull/547)
 - Add more validations to `RunnerTask`. [#554](https://github.com/Shopify/kubernetes-deploy/pull/554)
 - Validate secrets with `--server-dry-run` on supported clusters. [#553](https://github.com/Shopify/kubernetes-deploy/pull/553)
+- Add resource type `PersistentVolumeClaim` to prune whitelist. [#573](https://github.com/Shopify/kubernetes-deploy/pull/573)
```
Comment (KnVerey) [inline CHANGELOG.md]: This should explicitly be marked as a breaking change and include instructions on how to verify whether your cluster is affected and how to exempt resources from pruning. See https://github.com/Shopify/kubernetes-deploy/blob/master/CHANGELOG.md#0270 for a starting point.

Comment (KnVerey) [reply CHANGELOG.md]: Also I think you need to rebase. The changes showing in "next" above were already released.

Comment (jessie-JNing) [reply CHANGELOG.md]: Thanks @KnVerey , that's really a good remind. I'll add it as breaking change and include instructions for it

Comment (jessie-JNing) [reply CHANGELOG.md]: I think [Validate secrets with `--server-dry-run`](https://github.com/Shopify/kubernetes-deploy/pull/573/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR10) and [PersistentVolumeClaim in whitelist](https://github.com/Shopify/kubernetes-deploy/pull/573/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR12) should be in `0.29.0 enchantments`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,11 @@
 
 *Enchantments*
 - The KubernetesDeploy::RenderTask now supports a template_paths argument ([#555](https://github.com/Shopify/kubernetes-deploy/pull/546))
+- Validate secrets with `--server-dry-run` on supported clusters. ([#553](https://github.com/Shopify/kubernetes-deploy/pull/553))
 - We no longer hide errors from apply if all sensitive resources have passed server-dry-run validation. ([#570](https://github.com/Shopify/kubernetes-deploy/pull/570))
-
+- **[Breaking change]** Added PersistentVolumeClaim to the prune whitelist. ([#573](https://github.com/Shopify/kubernetes-deploy/pull/573))
```
Comment (KnVerey) [inline CHANGELOG.md]: 0.29.0 was already released. New stuff should be going under "next" above.

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,7 +27,6 @@
 - Officially support Kubernetes 1.15 ([#546](https://github.com/Shopify/kubernetes-deploy/pull/546))
 - Make sure that we only declare a Service of type LoadBalancer as deployed after its IP address is published. [#547](https://github.com/Shopify/kubernetes-deploy/pull/547)
 - Add more validations to `RunnerTask`. [#554](https://github.com/Shopify/kubernetes-deploy/pull/554)
-- Validate secrets with `--server-dry-run` on supported clusters. [#553](https://github.com/Shopify/kubernetes-deploy/pull/553)
```
Comment (KnVerey) [inline CHANGELOG.md]: Is it not true that this went into 0.28.0? 0.29.0 stopped suppressing errors, but this was the precursor PR.

Comment (jessie-JNing) [reply CHANGELOG.md]: Yes, this one is in `v0.28.0`. I'll put it back.

File: CHANGELOG.md (hunk 2 above)
Comment (jessie-JNing) [reply CHANGELOG.md]: Ah, OK, I was confused by this [compare](https://github.com/Shopify/kubernetes-deploy/compare/v0.29.0...master) .
I'll put it in `next` section

**Pull request merged**